cmake/Modules/FindSDL_mixer.cmake

```diff
@@ -26,13 +26,16 @@ module, but with modifications to recognize OS X frameworks and
 additional Unix paths (FreeBSD, etc).
 #]=======================================================================]
 
+unset(SDL_MIXER_LIBRARY CACHE)
```

Comment:
Don't do that. This will search again on every configure
I suggest using a variable for holding the prefix or whole varname (SDL_MIXER_LIBRARY vs SDL2_MIXER_LIBRARY)